lib/utils/CMakeLists.txt

```diff
@@ -0,0 +1 @@
+collect (PROJECT_LIB_SOURCES string.c)
```

Comment:
many RTOS support strlcpy need some method to remove string.c or strlcpy from build.

Reply:
Do you see a potential failure in not using this RTOS strlcpy implementation? 

perhaps defining is a weak function would answer to your concern?

Reply:
> Do you see a potential failure in not using this RTOS strlcpy implementation?
> 

the link will report the multiple instance error if RTOS provide one implementation.

> perhaps defining is a weak function would answer to your concern?

Yes, it's better.

Reply:
weak attribute added

Reply:
@arnopo can we introduce metal_strlcpy in libmetal and use it in open-amp instead?

That will resolve all duplicate name issues. Also, users can use default strlcpy (provided by OS or glibc).

 Please feel free to drop this idea if not suitable.

Reply:
yes that seems a good alternative to address the different implementations

Reply:
I like @tnmysh's idea, move metal_strcpy to libmetal/lib/utilities.h.

Reply:
I am quite hesitant about that, as this new code have no OS/processor/compiler dependency.
So I would avoid to create a new libmetal API for that.
I propose to keep it in openamp ( perhaps renaming string.x to utilities.x) and move it to libmetal later, if we need different implementations

Reply:
Sounds good. Yeah string.c should be renamed to utilities.c

Reply:
done

Reply:
OK